### PR TITLE
chore: add the string that failed validation to `IdentError::InvalidChars` error variant

### PR DESCRIPTION
**Pull request by greenhat**
## Describe your changes

Add the string that failed validation to `IdentError::InvalidChars` error variant to see the failed string in the error message.

## Checklist before requesting a review
- Repo forked and branch created from `next` according to naming convention.
- Commit messages and codestyle follow [conventions](./CONTRIBUTING.md).
- Relevant issues are linked in the PR description.
- Tests added for new functionality.
- Documentation/comments updated according to changes.

### CODE REVIEW

**Review by plafer — COMMENTED** [review]
LGTM, except for a minor change

File: assembly/src/ast/procedure/name.rs [hunk 1]
```diff
@@ -317,13 +317,14 @@ impl FromStr for ProcedureName {
                     '_' | '$' => false,
                     _ => true,
                 }) {
-                    Err(IdentError::InvalidChars)
+                    // Err(IdentError::InvalidChars { ident: s.into() })
+                    panic!("IdentError::InvalidChars")
```
Comment (plafer) [inline assembly/src/ast/procedure/name.rs:321]: I think you forgot to uncomment the `Err` here right? i.e. we shouldn't be panicking, and returning an error instead

Comment (bobbinth) [reply assembly/src/ast/procedure/name.rs:321]: @greenhat - let's fix this issue and merge. This way, we can include this PR into the next release of `miden-vm` (which I'm hoping to do on Monday).

Comment (greenhat) [reply assembly/src/ast/procedure/name.rs]: Thanks for catching this! Fixed. 
Sorry, I should've picked it up earlier. I botched my email filter for github notifications and missed it.

**Review by bobbinth — APPROVED** [review]
Looks good! Thank you!

**Pull request merged**